packages/orbit-components/src/Itinerary/ItinerarySegment/ItinerarySegmentDetail/index.tsx

```diff
@@ -87,7 +87,19 @@ const ItinerarySegmentDetail = ({
             ref={slideRef}
             tabIndex={-1}
             role="button"
-            onClick={toggleOpened}
+            onClick={ev => {
+              ev.stopPropagation();
+
+              const selection = window.getSelection();
+              const isTextSelected = selection?.toString() !== "";
+
+              if (!isTextSelected && ev.detail === 1) {
+                if (onCollapse) onCollapse(ev);
+                if (onClick) onClick(ev);
+              }
+
+              toggleOpened();
```

Comment:
I think we can avoid this logic here? 🤔 

On `ItinerarySegment` we already have the `toggleOpened` method definition. Can't we call `onCollapse` and `onExpand` there? We could even avoid the drill of the props here, no?

I didn't try this approach, so feel free to correct me if I suggested it wrong

Reply:
I will take a look 👀

Reply:
It looks like it's working ✨ I feel stupid for going in circles here, it was just right in front of my eyes 😅